builder/remotecontext/detect.go

```diff
@@ -129,11 +129,11 @@ func removeDockerfile(c modifiableContext, filesToRemove ...string) error {
 	case err != nil:
 		return err
 	}
+	defer f.Close()
```

Comment:
Looks like that's happening indeed;

```
06:41:21 time="2017-06-28T06:41:21Z" level=error msg="failed to remove .dockerignore: remove C:\\Users\\ContainerAdministrator\\AppData\\Local\\Temp\\builder-dockerignore-process-test250723583\\.dockerignore: The process cannot access the file because it is being used by another process." 
06:41:21 2017/06/28 06:41:21 Directory should contain exactly 1 file(s), got 2